File: go/S0009.go
package main

import "fmt"

func isPalindrome(x int) bool {
    if x < 0 {
        return false
    }
    origin := x
    redirect := 0
    for x != 0 {
        redirect = redirect*10 + x%10
        x /= 10
    }
    return origin == redirect
}

func main() {
    fmt.Println(isPalindrome(12321))
}

File: ts/S0009.ts
function isPalindrome(x: number): boolean {
    const str = String(x);
    let l = 0, r = str.length - 1;
    while (l < r) {
        if (str[l] !== str[r]) return false;
        l++, r--;
    }
    return true;
};

console.log(isPalindrome(-1221))
